app/full_text_search.py

```diff
@@ -113,51 +116,93 @@ def search(repo, query_txt, max_results):
     return results
 
 
-def create_jp_name_fields(**kwargs):
+def create_full_name_without_space(given_name, family_name):
```

Comment:
You may alternatively return a tuple: (given_name + family_name, family_name + given_name).

Otherwise, rename given_name, family_name here to name_component1, name_component2 or something, because currently family_name may be passed as given_name, which is confusing.